amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/RecordsRetrieved.java

```diff
@@ -24,4 +26,22 @@
      * @return the processRecordsInput received
      */
     ProcessRecordsInput processRecordsInput();
+
+    /**
+     * Returns the sequence number that can be used as a pagination token for next batch of records.
+     *
+     * @return sequenceNumber to checkpoint
+     */
+    default String batchSequenceNumber() {
```

Comment:
We dont use the term continuationSequenceNumber for polling API. For S2S it is the contSeqNum explicitly set and for Get it is the last successfully processed SeqNum in that record batch. Hence I wanted this to be a seqNum that uniquely identifies the record batch that was processed.

Reply:
As far as I see, these Ifaces are not integrated with Get path. The use of this attribute is continuationSequenceNumber, or checkpointSequenceNumber. 

the new batchSeqNum naming seems to be just another way of saying continuationSequenceNumber, and we should reuse same terms, instead of introducing new ones.

Reply:
Was referring to the following RecordsRetrieved Impl.

```
@Accessors(fluent = true)
    @Data
    static class PrefetchRecordsRetrieved implements RecordsRetrieved {

        final ProcessRecordsInput processRecordsInput;
        final String lastBatchSequenceNumber;
        final String shardIterator;

        PrefetchRecordsRetrieved prepareForPublish() {
            return new PrefetchRecordsRetrieved(processRecordsInput.toBuilder().cacheExitTime(Instant.now()).build(),
                    lastBatchSequenceNumber, shardIterator);
        }

    }
```

Reply:
Looking into it again, it makes sense to keep it as continuationSequenceNumber as RecordsRetrieved might still have this as a property for pagination.